### PR TITLE
CubeTexture: Added serialization/ deserialization

### LINKED ISSUE #9458: THREE.Texture.toJSON error when called on a material's envMap set to a cube map
I'm using THREE.js in a large application along with React and Redux. I have an object in the scene that has `mesh.material.envMap` set to a cube map, loaded through `THREE.CubeTextureLoader`.

Redux has a Chrome extension containing advanced dev tool features. Upon opening this extension, the following error appears:

`Uncaught TypeError: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The provided value is not of type '(HTMLImageElement or HTMLVideoElement or HTMLCanvasElement or ImageBitmap)'.`

I'm not certain what causes this error to appear, but since the dev tools open in a separate panel alongside the main application, it could be something to do with loss of focus.

The stack trace pointed me to the `getDataURL` function within `THREE.Texture.toJSON`, where the six images of the cube map are attempting to be stored:

```
if ( meta.images[ image.uuid ] === undefined ) {

        meta.images[ image.uuid ] = {
            uuid: image.uuid,
            url: getDataURL( image )
        };

}
```

This fails because `image` in this case is actually an array of 6 images, thus getDataURL is useless.

I was able to fix this with the following hack. Since I'm not sure what's precisely calling it, or what the actual side effects of such a fix are, I'm hesitant to submit it as a pull request. That said, I no longer see any errors with this fix in place.

```
if ( meta.images[ image.uuid ] === undefined ) {

    if (image instanceof Array && image.length > 0) {

        meta.images[ image.uuid ] = {
            uuid: image.uuid,
            url: image.map( (val) => { getDataURL(val) } )
        };

    } else {

        meta.images[ image.uuid ] = {
            uuid: image.uuid,
            url: getDataURL( image )
        };

    }

}
```

I have only tested this on r79 running Chrome on Windows 10, but I did also come across [this StackOverflow post](http://stackoverflow.com/questions/38733369/how-can-i-save-material-with-a-envmap-property), where I also submitted my fix.
##### Three.js version
- [x] r79
##### Browser
- [x] Chrome
##### OS
- [x] Windows

### PR DESCRIPTION
**Pull request by Mugen87**
Fixes #9458

### CODE REVIEW

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -532,12 +533,38 @@ Object.assign( ObjectLoader.prototype, {
...
+
+						var currentUrl = url[ j ];
+
+						var path = /^(\/\/)|([a-z]+:(\/\/)?)/i.test( currentUrl ) ? currentUrl : scope.texturePath + currentUrl;
+
+						images[ image.uuid ].push( loadImage( path ) );
+
+					}
+
+
```
Comment (takahirox) [inline src/loaders/ObjectLoader.js]: nit: Two empty lines here. We can remove one of them.

File: src/loaders/ObjectLoader.js [hunk 2]
```diff
@@ -532,12 +533,38 @@ Object.assign( ObjectLoader.prototype, {
...
+				} else {
 
-				images[ image.uuid ] = loadImage( path );
+					// load single image
+
+					var path = /^(\/\/)|([a-z]+:(\/\/)?)/i.test( image.url ) ? image.url : scope.texturePath + image.url;
+
+					images[ image.uuid ] = loadImage( path );
+
+
```
Comment (takahirox) [inline src/loaders/ObjectLoader.js]: nit: Same as the above.

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (Mugen87) [reply src/loaders/ObjectLoader.js]: 👍

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**